File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int rem, rev = 0;
        while (x != 0) {
            rem = x % 10;
            // Check for overflow before updating rev
            if (rev > Integer.MAX_VALUE / 10 || (rev == Integer.MAX_VALUE / 10 && rem > 7)) {
                return 0;
            }
            if (rev < Integer.MIN_VALUE / 10 || (rev == Integer.MIN_VALUE / 10 && rem < -8)) {
                return 0;
            }
            rev = (rev * 10) + rem;
            x = x / 10;
        }
        return rev;
    }
}

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        sign = -1 if x < 0 else 1
        x = abs(x)
        rev = 0
        while x != 0:
            rem = x % 10
            if rev > 2**31 // 10 or (rev == 2**31 // 10 and rem > 7):
                return 0
            if rev < -2**31 // 10 or (rev == -2**31 // 10 and rem < -8):
                return 0
            rev = (rev * 10) + rem
            x = x // 10
        return sign * rev